libevm/VM.cpp

```diff
@@ -233,6 +233,24 @@ void VM::interpretCases()
 		}
 		BREAK
 
+		CASE(REVERT)
+		{
+			// Pre-metropolis
+			if (!m_schedule->haveRevert)
+				throwBadInstruction();
+
+			m_newMemSize = memNeed(*m_SP, *(m_SP - 1));
+			updateMem();
+			ON_OP();
+			updateIOGas();
+
+			uint64_t b = (uint64_t)*m_SP--;
+			uint64_t s = (uint64_t)*m_SP--;
+			owning_bytes_ref output{ move(m_mem), b, s };
```

Comment:
Talk of a bigger refactoring reminds me that I've never liked the interface to the VM - the way it's a function pretending to be an object, the way that function takes its ration of gas by reference...  I attempted to eliminate the _io_gas reference once and gave up, as it had insinuated itself too deep in the code.